src/components/CanvasSaveButton/CanvasSaveButton.js

```diff
@@ -181,78 +206,135 @@ class CanvasSaveButton extends PureComponent {
     mapToExport.setTarget(null);
   }
 
-  getDownloadImageName() {
-    return (
-      `${window.document.title.replace(/ /g, '_').toLowerCase()}` +
-      `.${this.fileExt}`
+  drawCopyright(destContext, clip) {
+    const { extraData, scale } = this.props;
+    const { text, font, fillStyle } = extraData.copyright;
+    const copyright = typeof text === 'function' ? text() : text;
+
+    destContext.save();
+    destContext.scale(scale, scale);
+    // eslint-disable-next-line no-param-reassign
+    destContext.font = font || '12px Arial';
+    // eslint-disable-next-line no-param-reassign
+    destContext.fillStyle = fillStyle || 'black';
+    destContext.fillText(
+      copyright,
+      this.padding,
+      clip.h / scale - this.padding,
     );
+    destContext.restore();
   }
 
-  createCanvasImage(mapToExport) {
+  drawNorthArrow(destContext, clip) {
+    const { scale, extraData } = this.props;
+    const { src, circled, width, height, rotation } = extraData.northArrow;
+
     return new Promise(resolve => {
-      const { extent, scale, extraData, coordinates } = this.props;
+      const img = new Image();
+      if (src) {
```

Comment:
`img.src = src || circled ? NorthArrowCircle : NorthArrowSimple;`
Could be shorter.

Reply:
fixed